addons/crm/models/crm_lead.py

```diff
@@ -1164,57 +1164,74 @@ def get_rainbowman_message(self):
+                SUM(CASE WHEN source_id = %(source_id)s AND team_id = %(team_id)s THEN 1 ELSE 0 END) as count_source_closed_year,
+                MIN(CASE WHEN date_closed >= %(today)s - INTERVAL '30 days' AND team_id = %(team_id)s THEN day_close ELSE 1000000 END) as min_day_close_30,
+
+                SUM(CASE WHEN date_closed::date = (%(today)s - INTERVAL '1 day')  AND user_id = %(user_id)s THEN 1 ELSE 0 END) as count_user_closed_yesterday,
+                SUM(CASE WHEN date_closed::date = (%(today)s - INTERVAL '2 days') AND user_id = %(user_id)s THEN 1 ELSE 0 END) as count_user_closed_minus2day,
+                SUM(CASE WHEN date_closed::date = (%(today)s - INTERVAL '3 days') AND user_id = %(user_id)s THEN 1 ELSE 0 END) as count_user_closed_minus3day
             FROM crm_lead
             WHERE
-                type = 'opportunity'
+                type='opportunity'
             AND
                 active = True
             AND
                 probability = 100
             AND
-                DATE_TRUNC('year', date_closed) = DATE_TRUNC('year', CURRENT_DATE)
+                DATE_TRUNC('year', date_closed) = DATE_TRUNC('year', %(today)s)
             AND
                 (user_id = %(user_id)s OR team_id = %(team_id)s)
         """
-        self.env.cr.execute(query, {'user_id': self.user_id.id,
-                                    'team_id': self.team_id.id})
+        self.env.cr.execute(query, {
+            'user_id': self.user_id.id,
+            'team_id': self.team_id.id,
+            'country_id': self.country_id.id or -1,
+            'source_id': self.source_id.id or -1,
+            'today': fields.Date.context_today(self),
+        })
         query_result = self.env.cr.dictfetchone()
 
-        message = False
-        if query_result['total_won'] == 1:
-            message = _('Go, go, go! Congrats for your first deal.')
-        elif query_result['max_team_30'] == self.expected_revenue:
-            message = _('Boom! Team record for the past 30 days.')
-        elif query_result['max_team_7'] == self.expected_revenue:
-            message = _('Yeah! Deal of the last 7 days for the team.')
-        elif query_result['max_user_30'] == self.expected_revenue:
-            message = _('You just beat your personal record for the past 30 days.')
-        elif query_result['max_user_7'] == self.expected_revenue:
-            message = _('You just beat your personal record for the past 7 days.')
-        return message
+        if query_result['count_user_closed_year'] == 1:
+            return _('Go, go, go! Congrats for your first deal.')
+        elif self.expected_revenue and query_result['max_team_30'] == self.expected_revenue:
+            return _('Boom! Team record for the past 30 days.')
+        elif self.expected_revenue and query_result['max_team_7'] == self.expected_revenue:
+            return _('Yeah! Deal of the last 7 days for the team.')
```

Comment:
"Best" deal?
Or "Deal of the week"?
Sounds a bit odd to me but I guess to fight out with Luc perhaps